Userland/Libraries/LibWeb/CSS/ResolvedCSSStyleDeclaration.cpp

```diff
@@ -203,8 +216,68 @@ RefPtr<StyleValue const> ResolvedCSSStyleDeclaration::style_value_for_property(L
             value_or_default(maybe_background_origin, IdentifierStyleValue::create(CSS::ValueID::PaddingBox)),
             value_or_default(maybe_background_clip, IdentifierStyleValue::create(CSS::ValueID::BorderBox)));
     }
+    case CSS::PropertyID::BackgroundAttachment:
+        return style_value_for_background_property(
+            layout_node,
+            [](auto& layer) { return IdentifierStyleValue::create(to_value_id(layer.attachment)); },
+            [] { return IdentifierStyleValue::create(CSS::ValueID::Scroll); });
+    case CSS::PropertyID::BackgroundClip:
+        return style_value_for_background_property(
+            layout_node,
+            [](auto& layer) { return IdentifierStyleValue::create(to_value_id(layer.clip)); },
+            [] { return IdentifierStyleValue::create(CSS::ValueID::BorderBox); });
     case PropertyID::BackgroundColor:
         return ColorStyleValue::create(layout_node.computed_values().background_color());
+    case CSS::PropertyID::BackgroundImage:
+        return style_value_for_background_property(
+            layout_node,
+            [](auto& layer) -> NonnullRefPtr<StyleValue const> {
+                if (layer.background_image)
+                    return *layer.background_image;
+                return IdentifierStyleValue::create(CSS::ValueID::None);
+            },
+            [] { return IdentifierStyleValue::create(CSS::ValueID::None); });
+    case CSS::PropertyID::BackgroundOrigin:
+        return style_value_for_background_property(
+            layout_node,
+            [](auto& layer) { return IdentifierStyleValue::create(to_value_id(layer.origin)); },
+            [] { return IdentifierStyleValue::create(CSS::ValueID::PaddingBox); });
+    case CSS::PropertyID::BackgroundPosition:
```

Comment:
Should this include cases for `CSS::PropertyID::BackgroundPositionX` and `CSS::PropertyID::BackgroundPositionY` now?

Reply:
oh right, it should. thanks for noticing.
i will try to add them in the evening :)

Reply:
i removed background-position and background-size completely because they would require more work than just returning an unresolved value. i don't feel i could implement them now. sorry.